operator/cmd/configurator/fixup.go

```diff
@@ -0,0 +1,46 @@
+package configurator
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/cockroachdb/errors"
+	"sigs.k8s.io/yaml"
+
+	"github.com/redpanda-data/redpanda-operator/operator/pkg/clusterconfiguration"
+	"github.com/redpanda-data/redpanda-operator/operator/pkg/secrets"
+)
+
+func applyFixups[T any](ctx context.Context, config T, fn string, cloudExpander *secrets.CloudExpander) (T, error) {
+	var zero T
+	// It's harmless if the fixup file does not exist
+	fixupContent, err := os.ReadFile(fn)
+	var pe *os.PathError
+	if errors.As(err, &pe) {
+		return config, nil
+	}
+	if err != nil {
+		return zero, fmt.Errorf("cannot load fixup file %q: %w", fn, err)
+	}
+	var fs []clusterconfiguration.Fixup
+	if err := yaml.Unmarshal(fixupContent, &fs); err != nil {
+		return zero, fmt.Errorf("cannot unmarshal fixup file %q: %w", fn, err)
+	}
+	tpl := clusterconfiguration.Template[T]{
+		Content: config,
+		Fixups:  fs,
+	}
+	// Load the environment
+	env := make(map[string]string)
+	for _, ev := range os.Environ() {
+		name, val, found := strings.Cut(ev, "=")
+		if found {
+			env[name] = strings.Trim(val, "\n")
```

Comment:
Is this technically correct? I don't think anything bad will happen if we strip whitespace but `os.Environ()` shouldn't contain trailing newlines AFAIK.

Reply:
Turns out it can if it's populated from a configmap entry reference and the CM itself uses any kind of YAML serialisation other than `key: "the value is here"`.